src/Catalyst/Step6.tsx

```diff
@@ -151,6 +127,35 @@ const CopyButton = ({text}: {text: string}) => (
   </TouchableOpacity>
 )
 
+const useBlockGoBack = () => {
+  const navigation = useNavigation()
+
+  useEffect(() => {
+    const handleBeforeRemove = (
+      e: EventArg<
+        'beforeRemove',
+        true,
+        {
+          action: Readonly<{
+            type: string
+            payload?: object | undefined
+            source?: string | undefined
+            target?: string | undefined
+          }>
+        }
+      >,
+    ) => {
+      if (e.data.action.type !== 'RESET') {
+        e.preventDefault()
+      }
+    }
+    navigation.addListener('beforeRemove', handleBeforeRemove)
+    return () => {
+      navigation.removeListener('beforeRemove', handleBeforeRemove)
```

Comment:
i think removeListener is deprecated
`const unsubscribe = navigation.addListener(...)`

Reply:
the only thing this does is prevent going back, except for "reset"?

Reply:
also, there is a `listeners` prop for each screen, can we use that instead?
https://reactnavigation.org/docs/navigation-events/#listeners-prop-on-screen

Reply:
Hmm weird this example was added 8 days ago https://github.com/react-navigation/react-navigation/blob/4e4935ac2584bc1a00209609cc026fa73e12c10a/example/types.check.tsx#L91
`addListener` doesn't return anything (on types) https://github.com/react-navigation/react-navigation/blob/main/packages/core/src/types.tsx#L127

Reply:
> i think removeListener is deprecated `const unsubscribe = navigation.addListener(...)`

it looks like is not marked as deprecated, anyway updating to the unsub/sub made the job easier